_posts/2016-10-24-deploy-jekyll-with-travis.md

```diff
@@ -0,0 +1,232 @@
+[docs](https://docs.travis-ci.com/).
+
+Previously, Travis would build the site and run our tests, then we would merge
+branches into `master`, triggering GitHub Pages to rebuild and deploy our site.
+But we want to build our site before it gets to GitHub Pages to ensure our
+Jekyll Plugins are included. In our new workflow:
+
+1. We'll create a new default branch so the `master` branch can be cleared
+out and used by Travis.
+2. When this new branch is updated (i.e. a pull request is merged), Travis will
+build the site.
+3. Using a GitHub token for authorization, Travis will commit the `_site`
+directory to the `master` branch, then push the `master` branch.
+4. This will trigger GitHub Pages to deploy our pre-built site.
+
+### Create a new default branch
+
+We created a new branch off `master` called `source` and set it to the default
+branch on GitHub. Each time the deploy process is run, we'll have Travis delete
+the files on the `master` branch, then commit the entire compiled `_site`
+directory. This will ensure that all changes (including deletions) are captured.
+
+### Set up a GitHub token for Travis
+
+To give Travis the authority to commit and push to our repository, we generated
+a personal access token by visiting the
+[settings page](https://github.com/settings/tokens) on GitHub and generating a
+new token. We used a bot account that has access to our repo to avoid needing to
+use one of our personal accounts.
+
+![Generating a personal access token in GitHub]({{ site.url }}/assets/img/blog/github-token.jpg)
+
+We'll need to use the access token and the account's email address in our
+deployment script, so to keep those items out of version control we used
+`travis encrypt`:
+
+```bash
+$ travis encrypt GH_TOKEN=<token> --add env.global
+$ travis encrypt GH_EMAIL=<email> --add env.global
+```
+
+This adds encrypted versions of these sensitive items to `travis.yml`:
+
+```yaml
+env:
+  global:
+  - secure: [big long encrypted thing]
+  - secure: [and another]
+```
+
+With that, we can use the `GH_TOKEN` and `GH_EMAIL` variables in our
+deployment script.
+
+### Configure deployment
+
+Next we added the following to our `travis.yml` file:
+
+```yaml
+# Keep Travis from testing `master` branch
+branches:
+  except:
+  - master
+
+# Deployment config
+deploy:
+  provider: script
+  script: "./_scripts/build.sh"
+  skip_cleanup: false
+  on:
+    branch: source
+```
+
+We don't want Travis running our tests on the `master` branch anymore since it
+only contains the compiled site, and only want Travis to run the
+deployment script on the `source` branch. We're also telling Travis which script
+to run on deployment, and to destroy the site after the test script is run with our
+`config.test.yml` file so the deployment script can rebuild the site with our
+default `config.yml` file. Check out the [full travis.yml file here](https://github.com/savaslabs/savaslabs.github.io/blob/95bb1d70790acf4c33b7c121e2ff462af5096d73/.travis.yml).
```

Comment:
Also, maybe make it clear that it is an old revision of the file. Somethng like `Check out that revision of the full .travis.yml file`